src/core/control/ToolHandler.cpp

```diff
@@ -199,6 +199,8 @@ void ToolHandler::addToolChangedListener(ToolChangedCallback listener) {
 
 auto ToolHandler::getTool(ToolType type) const -> Tool& { return *(this->tools[type - TOOL_PEN]); }
 
+auto ToolHandler::getActiveTool() const -> Tool& {return *(this->activeTool); }
```

Comment:
This should return a pointer; if `activeTool` is null, this will segfault.

Reply:
True